src/network/networkd-util.c

```diff
@@ -113,6 +114,42 @@ DEFINE_STRING_TABLE_LOOKUP_FROM_STRING(dhcp_deprecated_address_family, AddressFa
 DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING(ip_masquerade_address_family, AddressFamily);
 DEFINE_STRING_TABLE_LOOKUP(dhcp_lease_server_type, sd_dhcp_lease_server_type_t);
 
+bool link_should_mark_config(Link *link, bool only_static, NetworkConfigSource source, uint8_t protocol) {
+        /* Always mark static configs. */
+        if (source == NETWORK_CONFIG_SOURCE_STATIC)
+                return true;
+
+        /* Ignore dynamically assigned configs. */
+        if (source != NETWORK_CONFIG_SOURCE_FOREIGN)
+                return false;
+
+        /* When 'only_static' is true, do not mark foreign configs. */
+        if (only_static)
+                return false;
```

Comment:
Shouldn't this check be done before the second check?

Reply:
There is no particular order for the second and third checks. But, maybe swapping the check makes the code easy to understand. So, swapped.